src/main/java/seedu/storage/Storage.java

```diff
@@ -0,0 +1,104 @@
+package storage;
+
+import seedu.duke.DukeException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final File file;
+    private final SimpleDateFormat ft = new SimpleDateFormat("dd MMM yyy h.mma");
+
+    public Storage() throws IOException {
+        Path filePath = Paths.get(System.getProperty("user.dir"), "duke.txt");
+        System.out.println(filePath);
+        if (!Files.exists(filePath)) {
+            Files.createFile(filePath);
+        }
+
+        this.file = new File(String.valueOf(filePath));
+    }
+
+    /**
+     * Converts the data in the file into a task list
+     * @return A task list containing task retrieved from file
+     * @throws IOException file error
+     * @throws ParseException date not correct format
+     */
+    public ArrayList<Task> read() throws IOException, ParseException {
+
+        Scanner sc = new Scanner(this.file);
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        while (sc.hasNext()) {
+            String[] task = sc.nextLine().trim().split("\t");
```

Comment:
```suggestion
            String[] taskSegments = sc.nextLine().trim().split("\t");
```
Perhaps you can nouns that reflect the variable better. Plural form is also be used when representing a collection of objects.